library/std/src/sys/windows/path/absolute.rs

```diff
@@ -0,0 +1,76 @@
+// Original implementation taken from crate "normpath".
+// Copyright 2020 "dylni" (https://github.com/dylni)
+
+use super::super::{c, fill_utf16_buf, os2path, to_u16s, IsZero};
+use crate::io;
+use crate::path::{Path, PathBuf};
+use crate::ptr;
+
+#[cfg(test)]
+mod tests;
+
+/// Calls `GetFullPathNameW` with a workaround for incorrect results.
+pub(crate) unsafe fn get_full_path_name(
+    lpfilename: *const u16,
+    buffer: *mut u16,
+    size: c::DWORD,
+) -> c::DWORD {
+    // While the docs for `GetFullPathNameW` have the standard note about
+    // needing a `\\?\` path for a long `lpfilename`, this does not appear to
+    // be true in practice.
+    // See:
+    // https://stackoverflow.com/questions/38036943/getfullpathnamew-and-long-windows-file-paths
+    // https://googleprojectzero.blogspot.com/2016/02/the-definitive-guide-on-win32-to-nt.html
+    let result = c::GetFullPathNameW(lpfilename, size, buffer, ptr::null_mut());
+    if result.is_zero() {
+        return result;
+    }
+
+    // WinAPI can recommend an insufficient size that causes it to return
+    // incorrect results, so extra space is reserved as a workaround:
+    // https://github.com/dylni/normpath/issues/5
+    let required_size = result.saturating_add(2);
+    if size < required_size { required_size } else { result }
+}
+
+/// Make a Windows path absolute.
+pub(crate) fn absolute(path: &Path) -> io::Result<PathBuf> {
+    let path = path.as_os_str();
+    // NULs in verbatim paths are rejected for consistency.
+    if path.bytes().contains(&0) {
+        return Err(io::const_io_error!(
+            io::ErrorKind::InvalidInput,
+            "strings passed to WinAPI cannot contain NULs",
+        ));
+    }
+
+    let prefix = super::parse_prefix(path);
+    // Verbatim paths should not be modified.
+    if prefix.map(|x| x.is_verbatim()).unwrap_or(false) {
+        let mut path = path.to_owned();
+        // SAFETY: The prefix only uses ASCII characters.
+        //
+        // Converting more of a verbatim path can change its meaning. The part
+        // changed at the start only indicates that the path is verbatim.
+        for ch in unsafe { &mut path.bytes_mut()[..4] } {
+            if ch == &b'/' {
+                *ch = b'\\';
+            }
+        }
+        return Ok(path.into());
```

Comment:
Hm, this can change the meaning of the path, no? If you pass a path that starts with `//?/` to `CreateFileW` then it will be parsed similarly to a device path and not be treated as a verbatim path. E.g. passing in `//?/C:/windows/system32/notepad.exe` will work but `\\?\C:/windows/system32/notepad.exe` won't.

Reply:
It seems Windows is not consistent about this. Running `dir "//?/C:\"`, for example, will return information for `\\?\C:\`. However, I agree that due to the inconsistency, changing the path is not a good idea.

This also might mean that changing `parse_prefix` would be a bad idea.